### DISCUSSION TITLE
Welcome to CAD_to_OpenMC Discussions!

### DISCUSSION THREAD

**Discussion by ebknudsen** [united-neux/CAD_to_OpenMC#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JoseNogueron** [comment]
Hi! I’m fairly new to OpenMC. I’m trying to import a CAD model created in CATIA using DAGMC. I’m generating an .h5m file from a very simple geometry made up of two hexahedral parts. In the STEP file, I’ve given each solid a different name, but I can’t get materials assigned so that OpenMC recognizes them. Do the material tags need to be defined before converting the STEP file to H5M, or can they be added after importing the H5M?
This is the summary I’m getting:

SUMMARY: solid_id material_tag     stl-file
SUMMARY:        1 vacuum           volume_1.stl
SUMMARY:        2 vacuum           volume_2.stl

I would appreciate any kind of help. 
Thank you so much, 
Jose

**Reply by ebknudsen** [reply]
Hi Jose,
Great that you're using CAD_to_OpenMC, less so that you have problems, but let's see what what we can do.

The way this is supposed to work is that the part/solid names are supposed to be used as material tags in the resulting h5m-file.
For some reason this does not happen in your case, instead the parts are tagged with the default material: vacuum.
To help you better, I'd need a bit more information however. Could you supply me with:
a. your CAD_to_OpenMC-script
b. the step-file
c. the resulting h5m-file

If you for some reason are not allowed to share any or all of these, you can perhaps explain in detail what calls to CAD_to_OpenMC you issue.
cheers
Erik

**Reply by JoseNogueron** [reply]
Hi Erik! 
The parts in the CAD file are created with the names of the supposed materials: FW and BB. 
And the code used to generate the h5m file is practically the same I found in the examples. 
Attached you will find the stp file and the code to create the h5m model that I use. 

Thank you so much, 
[Model_Hexa.zip](https://github.com/user-attachments/files/22623277/Model_Hexa.zip)

Jose

**Reply by ebknudsen** [reply]
Thanks - I'll take a look. Stay tuned
